data-prepper-plugins/mutate-string-processors/README.md

```diff
@@ -0,0 +1,106 @@
+# Mutate String Processors
+The following is a list of processors to mutate a string.
+
+## SubstituteStringProcessor
+A processor that matches a key's value against a regular expression and replaces all matches with a replacement string.
+
+### Example
+The following `substitute_string` processor example will replace all ':' with '-' for the `message` key.
+
+```yaml
+processor:
+  - substitute_string:
+      entries:
+        - source: "message"
+          from: ":"
+          to: "-"
+```
+If `from` regex string does not have a match, the key will be returned as it is.
+
+### Configuration
+* `entries` - (required) - A list of entries to add to an event
+    * `source` - (required) - The key to be modified
+    * `from` - (required) - The Regex String to be replaced
+    * `to` - (required) - The String to be substituted for each match of `from`
+    
+---
+
+## SplitStringProcessor
+A processor that splits a field into an array using a delimiter character.
+
+### Example
+The following `split_string` processor example will split the key `"hello,world"` into `[hello, world]` array.
+
+```yaml
+processor:
+  - split_string:
+      entries:
+        - source: "hello,world"
+          delimiter: ","
+```
+
+### Configuration
+* `entries` - (required) - A list of entries to add to an event
+    * `source` - (required) - The key to be split
+    * `delimiter` - (required) - The separator character responsible for the split
+
+---
+
+## UppercaseStringProcessor
+A processor that converts a key to its uppercase counterpart.
+
+### Example
+The following `uppercase_string` processor example will convert the key to its uppercase counterpart.
+
+```yaml
+processor:
+  - uppercase_string:
+      with_keys:
+        - "uppercaseField"
+```
+
+### Configuration
+* `with_keys` - (required) - A list of keys to convert to Uppercase
+
+---
+
+## LowercaseStringProcessor
+A processor that converts a string to its lowercase counterpart.
+
+### Example
+The following `lowercase_string` processor example will convert the key to its lowercase counterpart.
+
+```yaml
+processor:
+  - lowercase_string:
+      with_keys:
+        - "lowercaseField"
+```
+
+### Configuration
+* `with_keys` - (required) - A list of keys to convert to Lowercase
+
+---
+
+## TrimStringProcessor
+A processor that strips whitespace from a string.
```

Comment:
I believe it trims from the beginning and end only. Please specify that in the documentation.